### ISSUE TITLE
java tooling support

### ISSUE THREAD

**Issue by jvmlet** [kohsuke/winsw#150]
Hi
I've successfully  deployed the executable  jar as windows service using below configuration

````
<executable>java</executable>
<arguments>-Xmx1024m -jar myJar.jar  myConfig.yml</arguments>
````

When I start VisualVM/JConsole/MissionControl  locally,  I don't see the java process listed in these tools.
When I run  `java -Xmx1024m -jar myJar.jar  myConfig.yml` from command line, everything works as expected.

Please advise,
Thanks

**Comment by oleg-nenashev** [comment]
> When I start VisualVM/JConsole/MissionControl locally, I don't see the java process listed in these tools.

In order to connect to the process running as a service, you firstly need to have proper permissions. E.g. if your service runs under the system account, you could try running VisualVM as an administrator.

For VisualVM it is also a requirement to run VisualVM and the app process with the same JDK IIRC.

**Comment by jvmlet** [comment]
I've configured the service to run under account from Domain Controller 
![image](https://cloud.githubusercontent.com/assets/6664858/20925905/48c25df4-bbc2-11e6-93be-2dc47f6bf91c.png)
and I'm logged in to the system with the same account.
  Running the Visual VM as admin doesn't solve the issue ...

**Comment by oleg-nenashev** [comment]
Do you use the same JDK?

**Comment by jvmlet** [comment]
Yes, sure, when I run `java -Xmx1024m -jar myJar.jar myConfig.yml` from command line I can see the process in JConsole

**Comment by oleg-nenashev** [comment]
Will try to reproduce later.

**Comment by ultimoamore** [comment]
any news? I'm having the exact same problem

**Comment by rstirling** [comment]
Hope it helps: https://blogs.oracle.com/nbprofiler/monitoring-java-processes-running-as-a-windows-service

**Comment by jvmlet** [comment]
Thanks @rstirling, it does.

**Comment by zjphdh2604** [comment]
i found a tool named arthas devloped by alibaba , not detected because the different classloader , you can use CMD classload found the jvm classloader tree hashcode ,then use the CMD like ognl -c #hashcode  #ognl detail, you can found out the object instance you want. https://alibaba.github.io/arthas/commands.html

**Comment by fantasy0v0** [comment]
> Thanks @rstirling, it does.

Have you solved this problem？I can't open this link。😣

**Comment by duemir** [comment]
For the record. Web Archive link of the previously mentioned article

https://web.archive.org/web/20200919225906/https://blogs.oracle.com/nbprofiler/monitoring-java-processes-running-as-a-windows-service
